staging/src/k8s.io/apiserver/pkg/storage/tests/cacher_test.go

```diff
@@ -692,3 +692,85 @@ func TestRandomWatchDeliver(t *testing.T) {
 		watched++
 	}
 }
+
+func TestCacherListerWatcher(t *testing.T) {
```

Comment:
Have you considered / ruled-out an integration test as well?  Off the cuff I couldn't think of anything that would be effective in that mode, but was curious whether I'm just missing something.

Reply:
I couldn't find a good way to directly verify chunking of reflector via an integration since chunking is hidden from the caller. 

But I'm poking around at the integration tests, and I can't much for "watch cache disabled" behavior, so I can improve that so we test a reflector against a watch cache disabled apiserver. That would give us black box coverage for the reflector part of this change.

Reply:
I've added two more tests: unit test for chunked listing in the reflector, integration test for pager based listing for RV=0, both with watch cache enabled and disabled. Still no integration test specifically covering the functionality of this change, but all the constitute parts are now well covered and existing e2e and integration tests indirectly cover the end-to-end code paths.